L1Trigger/TrackFindingTracklet/test/ProducerTBout.cc

```diff
@@ -0,0 +1,184 @@
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/Framework/interface/Run.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Utilities/interface/EDGetToken.h"
+#include "FWCore/Utilities/interface/EDPutToken.h"
+#include "FWCore/Utilities/interface/ESGetToken.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "DataFormats/Common/interface/Handle.h"
+
+#include "L1Trigger/TrackTrigger/interface/Setup.h"
+#include "L1Trigger/TrackerTFP/interface/DataFormats.h"
+#include "L1Trigger/TrackFindingTracklet/interface/ChannelAssignment.h"
+#include "L1Trigger/TrackFindingTracklet/interface/Settings.h"
+
+#include <string>
+#include <vector>
+#include <deque>
+#include <iterator>
+#include <cmath>
+#include <numeric>
+
+using namespace std;
+using namespace edm;
+using namespace trackerTFP;
+using namespace tt;
+
+namespace trklet {
+
+  /*! \class  trklet::ProducerTBout
+   *  \brief  Transforms TTTracks from Tracklet pattern reco. into f/w comparable format
```

Comment:
This comment seems misleading. I believe this module:
1) Adds to the digitised track stream a reference to the corresponding TTTrack (which could not be done in L1FPGATrackProducer, as the TTTrack hadn't yet been created).
2) Add TrackBuilder truncation to the digitized stub stream. And splits the stub stream into two streams depending on whether or not they passed the truncation cut. (These steps could theoretically have been done inside L1FPGATrackProducer).

Reply:
done

Reply:
Please add to comment my point 1 (if true). It helps understanding what this module does.

Reply:
I added this comment to code